core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala

```diff
@@ -2523,19 +2522,29 @@ class PlaintextAdminIntegrationTest extends BaseAdminIntegrationTest {
   def testCreateTopicsReturnsConfigs(quorum: String): Unit = {
     client = Admin.create(super.createConfig)
 
-    val alterMap = new util.HashMap[ConfigResource, util.Collection[AlterConfigOp]]
-    alterMap.put(new ConfigResource(ConfigResource.Type.BROKER, ""), util.Arrays.asList(
-      new AlterConfigOp(new ConfigEntry(KafkaConfig.LogRetentionTimeMillisProp, "10800000"), OpType.SET)))
-    (brokers.map(_.config) ++ controllerServers.map(_.config)).foreach { case config =>
-      alterMap.put(new ConfigResource(ConfigResource.Type.BROKER, config.nodeId.toString()),
-        util.Arrays.asList(new AlterConfigOp(new ConfigEntry(
-          KafkaConfig.LogCleanerDeleteRetentionMsProp, "34"), OpType.SET)))
+    val newLogRetentionProperties = new Properties
+    newLogRetentionProperties.put(KafkaConfig.LogRetentionTimeMillisProp, "10800000")
+    TestUtils.incrementalAlterConfigs(null, client, newLogRetentionProperties, perBrokerConfig = false)
+      .all().get(15, TimeUnit.SECONDS)
+
+    val newLogCleanerDeleteRetention = new Properties
+    newLogCleanerDeleteRetention.put(KafkaConfig.LogCleanerDeleteRetentionMsProp, "34")
+    TestUtils.incrementalAlterConfigs(brokers, client, newLogCleanerDeleteRetention, perBrokerConfig = true)
+      .all().get(15, TimeUnit.SECONDS)
+
+    if (isKRaftTest()) {
+      ensureConsistentKRaftMetadata()
+    } else {
+      waitUntilTrue(() => brokers.forall(_.config.originals.getOrDefault(
```

Comment:
Do we not want to verify propagation for kraft as well? I think `ensureConsistentKRaftMetadata` only ensures brokers have caught up.

Reply:
Please correct me if I am wrong here. 

My understanding is that `ensureConsistentKRaftMetadata` should ensure that the brokers have caught with the metadata available in kraft controllers. For alter config in this scenario, we are sure that one of the controller server contains the new configuration because we wait for the AlterConfig future to complete. Controller should have stored the new configuration in its metadata log before completing the future and `ensureConsistentKRaftMetadata` will ensure that all brokers are consistent with the controller.

Also note that other tests such as `testAppendConfig`, `testInvalidIncrementalAlterConfigs` and many more rely on `ensureConsistentKRaftMetadata` to ensure propagating before asserting in test.

Reply:
Fair enough. The assertions below do verify the overridden properties in the CreateTopics response, which is the point of the test.

Reply:
I am resolving this comment. Please let me know if I missed changing something here.